qiskit_nature/properties/second_quantization/electronic/dipole_moment.py

```diff
@@ -54,6 +58,33 @@ def __init__(
         name = self.__class__.__name__ + axis.upper()
         super().__init__(name, electronic_integrals, shift=shift)
 
+    def to_hdf5(self, parent: h5py.Group) -> None:
+        """Stores this instance in a HDF5 group inside of the provided parent group.
+
+        Args:
+            parent: the parent HDF5 group.
+        """
+        super().to_hdf5(parent)
+        group = parent.require_group(self.name)
+
+        group.attrs["axis"] = self._axis
+
+    @classmethod
+    def from_hdf5(cls, h5py_group: h5py.Group) -> DipoleMoment:
+        """Constructs a new instance from the data stored in the provided HDF5 group.
+
+        Args:
+            h5py_group: the HDF5 group from which to load the data.
+
+        Returns:
+            A new instance of this class.
+        """
+        integral_property = super().from_hdf5(h5py_group)
+
+        axis = h5py_group.attrs["axis"]
+
+        return cls(axis, list(integral_property), shift=integral_property._shift)
```

Comment:
You're right that this is not 100% consistent right now. I will update the code to explicitly use the correct class where possible.

Reply:
Hmm.. In some places it is not possible to use `cls()` because a sub-class could have a mismatching `__init__` function signature (e.g. `GroupedProperty`). Would you argue to never use `cls()` and always encode the class explicitly or would you prefer to use `cls()` wherever possible?

Reply:
If cls is used used it becomes a static method? where the class is coded explicitly. In that case I guess they would all be consistent, unlike now where some use cls and some don't. At first sight to the code there is this difference with some using and some not and I guess it takes extra effort for someone to figure things. If they are all consistent then its simpler I think. If there is good reason for how it is then it is what it is. I commented from being a newcomer and seeing it and seeing differences and cls being used or not for the method.

Reply:
Okay that makes sense. I will see if I can hard-code the class everywhere and make this a staticmethod instead of classmethod. If I run into issues with that, I will fix those and write a comment why certain cases require `cls` :+1:

Reply:
I just changed it to be `@staticmethod` :+1: